lib/block_view_access_groups.go

```diff
@@ -0,0 +1,1026 @@
+					"_connectAccessGroupCreate: Can't mute a member that's already muted")
+			}
+
+			// Add member to newMuteList
+			newMuteList = append(newMuteList, newlyMutedMember)
+		}
+
+	case AccessGroupOperationUnmuteMembers:
+		// Unmuting members assumes the group was already created.
+		if existingEntry == nil || existingEntry.isDeleted {
+			return 0, 0, nil, errors.Wrapf(RuleErrorAccessGroupDoesntExist,
+				"_connectAccessGroupCreate: Can't mute members for a non-existent group")
+		}
+
+		for _, newlyUnmutedMember := range txMeta.AccessGroupMembers {
+
+			// Make sure we are unmuting a member that exists in the group.
+			member := bav.GetMessagingMember(newlyUnmutedMember.GroupMemberPublicKey, existingEntry.GroupOwnerPublicKey, existingEntry.AccessGroupKeyName, blockHeight)
+			if member == nil {
+				return 0, 0, nil, errors.Wrapf(RuleErrorAccessMemberNotInGroup,
+					"_connectAccessGroupCreate: Can't unmute a non-existent member (%v)", newlyUnmutedMember.GroupMemberPublicKey[:])
+			}
+
+			// GroupOwner unmuting herself is invalid because GroupOwner can never be muted in the first place
+			if reflect.DeepEqual(newlyUnmutedMember.GroupMemberPublicKey[:], existingEntry.GroupOwnerPublicKey[:]) {
+				return 0, 0, nil, errors.Wrapf(RuleErrorAccessGroupOwnerUnmutingSelf,
+					"_connectAccessGroupCreate: GroupOwner cannot mute herself (%v).", existingEntry.GroupOwnerPublicKey[:])
+			}
+
+			// Get group member attribute entry for this member.
+			attributeEntry, err := bav.getGroupMemberAttributeEntry(existingEntry.GroupOwnerPublicKey, existingEntry.AccessGroupKeyName, newlyUnmutedMember.GroupMemberPublicKey, AccessGroupMemberAttributeIsMuted)
+			if err != nil {
+				return 0, 0, nil, errors.Wrapf(err, "_connectAccessGroupCreate: Problem getting group member attribute entry")
+			}
+			// Check if the member is already unmuted.
+			if attributeEntry == nil || !attributeEntry.IsSet {
+				return 0, 0, nil, errors.Wrapf(RuleErrorAccessMemberAlreadyUnmuted,
+					"_connectAccessGroupCreate: Can't unmute a member that's already unmuted")
+			}
+
+			// Add member to newUnmuteList
+			newUnmuteList = append(newUnmuteList, newlyUnmutedMember)
+		}
+
+	default:
+		// If we're here, then the operation type is invalid. Currently, this can only
+		// happen if the operation is of type AccessGroupOperationRemoveMembers
+		return 0, 0, nil, errors.Wrapf(err,
+			"_connectAccessGroupCreate: Error, messaging group operation.")
+
+	}
+
+	// merge extra data
+	var extraData map[string][]byte
+	if blockHeight >= bav.Params.ForkHeights.ExtraDataOnEntriesBlockHeight {
+		var existingExtraData map[string][]byte
+		if existingEntry != nil && !existingEntry.isDeleted {
+			existingExtraData = existingEntry.ExtraData
+		}
+		extraData = mergeExtraData(existingExtraData, txn.ExtraData)
+	}
+
+	// TODO: Currently, it is technically possible for any user to add *any other* user to *any group* with
```

Comment:
How can the app level tell if it's a garbage encrypted key? We should probably have some guards in place to prevent this junk

Reply:
So right now the simple solution could be having a member attribute called "AcceptedInvitation" which denotes whether a member of a group accepts to be a part of it or now. 

This prevent spam big time, otherwise random channels could keep adding people.

This would also prevent garbage EncryptedKeys because they would decrypt garbage messages. The member can reject group invitation at this point due to unreadability of messages.

Reply:
What do you think of this current solution @lazynina ? I'm not sure how the app layer would determine the correctness of an EncryptedKey. The burden currently lies on the user.